.github/workflows/upload-to-pypi.yml

```diff
@@ -0,0 +1,104 @@
+on:
+  workflow_dispatch:
+    inputs:
+      ansible-version:
+        description: >-
+          Release Version. Example : 11.1.0
+        required: true
+      ansible-major-version:
+        description: 'Ansible major version. Example: 11'
+        required: true
+
+env:
+  ANSIBLE_VERSION: "${{ inputs.ansible-version }}"
+
+# Build job builds the wheel and the source tarball and upload the artifacts
+
+jobs:
+  build:
+    name: Build sdist and wheel
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out antsibull
+        uses: actions/checkout@v3
+        with:
+          repository: ansible-community/antsibull
+          path: antsibull
+
+      - name: Pre-create build directory
+        run: |
+          mkdir -p antsibull/build
+
+      - name: Check out ansible-build-data
+        uses: actions/checkout@v3
+        with:
+          path: antsibull/build/ansible-build-data
+
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install dependencies
+        working-directory: antsibull
+        run: |
+          python3 -m pip install ansible-core antsibull
+          ansible-galaxy collection install community.general
+
+      - name: Create sdist and wheel using the deps file created in the ansible-build-data Release PR
+        working-directory: antsibull
+        run: |
+          ansible-playbook -vv playbooks/build-single-release.yaml -e "antsibull_ansible-version=${ANSIBLE_VERSION}" -e antsibull_data_reset=false
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: sdist-and-wheel
+          path: antsibull/build/ansible-*.*
+
+# publish job downloads the arifacts and publish it to PyPI
+
+  publish:
+    needs: build
+    name: 'Upload Ansible (${{ inputs.ansible-version }}) to PyPI'
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/project/ansible
+    permissions:
+      id-token: write
+
+    steps:
+
+    - name: Download artifact
+      uses: actions/download-artifact@v3
+      with:
+        name: sdist-and-wheel
+        path: dist/
+
+    - name: Upload Ansible sdist and wheel to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+
+# git-tag job creates the git tag
+
+  git-tag:
+    needs: publish
+    name: 'Creates git tag for Ansible (${{ inputs.ansible-version }})'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    steps:
+
+      - name: Check out ansible-build-data
+        uses: actions/checkout@v3
+
+      - name: Create git tag
+        working-directory: ansible-build-data
+        run: |
+          git config --global user.name "Github Actions"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git tag -a "${ANSIBLE_VERSION}" -m "Ansible ${ANSIBLE_VERSION}: Changelog, Porting Guide and Dependent Collection Details"
```

Comment:
Looks like this tags whatever's the last commit on the main branch. So if there's a race condition with something that manages squeeze in after the PR merge but before running this workflow, some other commit may end up being tagged, not the immediate PR merge. To combat this, we should probably track the merge commit associated with the PR.

Reply:
Yes, this will be a future improvement.